components/BlurhashImage.vue

```diff
@@ -57,81 +62,82 @@ export default Vue.extend({
-            return this.item.ImageBlurHashes?.Primary?.[
-              this.item?.ImageTags?.Primary
-            ];
-          }
-          break;
-        case ImageType.Thumb:
-          if (
-            this.item?.ImageTags?.Thumb &&
-            this.item.ImageBlurHashes?.Thumb?.[this.item?.ImageTags?.Thumb]
-          ) {
-            return this.item.ImageBlurHashes?.Thumb?.[
-              this.item?.ImageTags?.Thumb
-            ];
-          }
-        // eslint-disable-next-line no-fallthrough
-        case ImageType.Backdrop || ImageType.Thumb:
-          if (
-            this.item?.BackdropImageTags?.[0] &&
-            this.item?.ImageBlurHashes?.Backdrop
-          ) {
-            return this.item?.ImageBlurHashes?.Backdrop?.[
-              this.item?.BackdropImageTags?.[0]
-            ];
-          }
-      }
-      return '';
     },
-    checkImage(): boolean {
+    isValidBlurhash(): boolean {
       if (
-        (this.imageType === ImageType.Primary &&
-          this.item.ImageTags &&
-          this.item.ImageTags.Primary) ||
-        (this.imageType === ImageType.Backdrop &&
-          this.item.ImageTags &&
-          this.item.BackdropImageTags) ||
-        (this.imageType === ImageType.Thumb &&
-          this.item.ImageTags &&
-          this.item.ImageTags.Thumb)
+        excludedTypes.includes(this.type) ||
+        (this.item?.ImageBlurHashes as Array<never>).length === 0
+      ) {
+        return false;
+      } else if (
+        Object.prototype.hasOwnProperty.call(
+          this.item?.ImageBlurHashes,
+          this.type
+        ) &&
+        this.type !== ImageType.Backdrop
       ) {
         return true;
-      } else return false;
-    }
-  },
-  mounted(): void {
-    if (this.checkImage) {
-      const card = this.$refs.card as HTMLElement;
-      this.image = this.getImageUrlForElement(this.imageType, {
-        item: this.item,
-        element: card
+      } else if (
+        this.type === ImageType.Backdrop &&
+        (this.item?.BackdropImageTags as Array<never>).length > 0 &&
+        this.item?.ImageBlurHashes?.Backdrop &&
+        Object.prototype.hasOwnProperty.call(
+          this.item?.ImageBlurHashes?.Backdrop as Record<string, string>,
+          (this.item?.BackdropImageTags as Array<string>)[0]
+        )
+      ) {
+        return true;
+      } else {
+        return false;
+      }
+    },
+    getHash(): string {
+      if (this.type === ImageType.Backdrop) {
+        const tag = (this.item?.BackdropImageTags as Array<string>)[0];
+
+        return ((this.item?.ImageBlurHashes as BaseItemDtoImageBlurHashes)
+          .Backdrop as Record<string, string>)[tag];
+      } else {
+        return ((this.item?.ImageBlurHashes as BaseItemDtoImageBlurHashes)[
+          this.type
+        ] as Record<string, string>)[
+          (this.item?.ImageTags as Record<string, string>)[this.type]
+        ];
+      }
+    },
+    onError(): void {
+      this.pushSnackbarMessage({
```

Comment:
We shouldn't output a message for every error. Some, like this one, can just deal with an even sent back up.